CodeSniffer/Standards/Squiz/Sniffs/Commenting/FunctionCommentSniff.php

```diff
@@ -650,10 +650,12 @@ protected function processParams($commentStart, $commentEnd)
                     } else if (count($typeNames) === 1) {
                         // Check type hint for array and custom type.
                         $suggestedTypeHint = '';
-                        if (strpos($suggestedName, 'array') !== false) {
+                        if (strpos($suggestedName, 'array') !== false || substr($suggestedName, -2) === '[]') {
```

Comment:
If there is a | in the type, count($typeNames) will be greater than 1, and no type hint will be suggested. In other words: this piece of code doesn't have to cover composite types.